core/trino-parser/src/main/java/io/trino/sql/tree/CharLiteral.java

```diff
@@ -41,7 +39,7 @@ public CharLiteral(Optional<NodeLocation> location, String value)
         super(location);
         requireNonNull(value, "value is null");
         this.length = value.length();
-        this.value = CharMatcher.is(' ').trimTrailingFrom(value);
```

Comment:
@findepi 
want to learn from you, i do not know why we need to trimRight for char?
so
```
SELECT translate(CHAR 'A BC ', ' ', '_');
we transfer to in prepareQuery
SELECT translate(CHAR 'A BC', ' ', '_');
```
so could we remove this.

Reply:
The char type knows its (exactl) length, so the _internal_ representation of char values has trailing spaces omitted to save space.
From end-user perspective this space-related optimization is invisible.

Reply:
@findepi Thanks for sharing this, I went a wrong direction at the beginning. But finally I got the right solution, I think. :). And i learned a lot during this.
Thanks